commit 566b92ac4746d4c59eab14112efdc31737b922c7
Author: Arjen Hiemstra <ahiemstra@heimr.nl>
Date:   Thu May 14 22:35:52 2026 +0200

    Make an option for toggling using corrosion for bridge building
    
    Rather than only switching based on whether we're building a flatpak or
    not.

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f91dbc6..536fb92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ if (BUILD_DOCS)
     include(ECMGenerateQDoc)
 endif()
 
+option(USE_CORROSION_CXXBRIDGE "Use corrosion to build the CXX bridge" ON)
+
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_AUTOMOC FALSE)
 set(CMAKE_AUTORCC FALSE)
diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 16c46a0..a81165b 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -28,10 +28,10 @@ target_sources(cxx-rust-cssparser PUBLIC
     ${CMAKE_CURRENT_BINARY_DIR}/cssparser_export.h
 )
 
-if(DEFINED ENV{FLATPAK_ID})
-    target_link_libraries(cxx-rust-cssparser PRIVATE cxx_rust_cssparser_impl)
-else()
+if (USE_CORROSION_CXXBRIDGE)
     target_link_libraries(cxx-rust-cssparser PRIVATE cxx-rust-cssparser-impl-bridge)
+else()
+    target_link_libraries(cxx-rust-cssparser PRIVATE cxx_rust_cssparser_impl)
 endif()
 
 target_include_directories(cxx-rust-cssparser PUBLIC
diff --git a/rust/CMakeLists.txt b/rust/CMakeLists.txt
index 594abb1..8f9f3d5 100644
--- a/rust/CMakeLists.txt
+++ b/rust/CMakeLists.txt
@@ -3,7 +3,14 @@
 
 corrosion_import_crate(MANIFEST_PATH Cargo.toml)
 
-if(DEFINED ENV{FLATPAK_ID})
+if (USE_CORROSION_CXXBRIDGE)
+    corrosion_add_cxxbridge(cxx-rust-cssparser-impl-bridge
+        CRATE cxx_rust_cssparser_impl
+        FILES ffi.rs
+    )
+
+    set_target_properties(cxx-rust-cssparser-impl-bridge PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
+else()
     # Flatpak: use cxx_build output directly instead of corrosion_add_cxxbridge.
     string(SHA1 _cargo_path_hash ${CMAKE_CURRENT_SOURCE_DIR}/Cargo.toml)
     string(SUBSTRING "${_cargo_path_hash}" 0 5 _cargo_path_hash)
@@ -17,13 +24,6 @@ if(DEFINED ENV{FLATPAK_ID})
     )
 
     target_include_directories(cxx_rust_cssparser_impl INTERFACE ${_bridge_include_dir})
-else()
-    corrosion_add_cxxbridge(cxx-rust-cssparser-impl-bridge
-        CRATE cxx_rust_cssparser_impl
-        FILES ffi.rs
-    )
-
-    set_target_properties(cxx-rust-cssparser-impl-bridge PROPERTIES POSITION_INDEPENDENT_CODE TRUE)
 endif()
 
 if (BUILD_TESTING)
